torch/csrc/Device.cpp

```diff
@@ -137,6 +137,24 @@ PyObject *THPDevice_rc(PyObject *a, PyObject *b, int op) {
   END_HANDLE_TH_ERRORS
 }
 
+PyObject *THPDevice_reduce(THPDevice *self)
+{
+  auto ret = THPObjectPtr{PyTuple_New(2)};
+  if (!ret) throw python_error();
+
+  py::object torch_module = py::module::import("torch");
+  py::object torch_device = torch_module.attr("device");
```

Comment:
You need to wrap the code in `HANDLE_TH_ERRORS`, `END_HANDLE_TH_ERRORS` or otherwise those lines may throw an exception, and it will terminate the whole process since there's nothing to catch it.